samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -143,14 +171,18 @@ public Startpoint readStartpoint(SystemStreamPartition ssp, TaskName taskName) {
     Preconditions.checkState(!stopped, "Underlying metadata store not available");
```

Comment:
Would be a better to return Optional<Startpoint> here, rather than handling null at the caller-end.

Reply:
Sure